src/main/java/com/blobExample/client/ClientResource.java

```diff
@@ -0,0 +1,130 @@
+package com.blobExample.client;
+
+import com.blobExample.models.ClientRequest;
+import com.blobExample.models.ClientResponse;
+import com.blobExample.models.ServerResponse;
+import com.codahale.metrics.annotation.Timed;
+import com.expedia.blobs.core.*;
+import com.expedia.blobs.core.BlobStore;
+import com.expedia.blobs.core.Blobs;
+import com.expedia.blobs.core.BlobsFactory;
+import com.expedia.blobs.core.predicates.BlobsRateLimiter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.ObjectOutputStream;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+import javax.ws.rs.*;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.Invocation;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+
+@Path("/displayMessage")
+@Produces(MediaType.APPLICATION_JSON)
+public class ClientResource {
+
+    private final String template;
+    private final String defaultName;
+    private final AtomicLong counter;
+    private javax.ws.rs.client.Client jerseyClient;
+    private BlobStore blobStore;
+    private static final Logger LOGGER = LoggerFactory.getLogger(ClientResource.class);
+
+    public ClientResource(String template, String defaultName, Client jerseyClient, BlobStore blobStore) {
+        this.template = template;
+        this.defaultName = defaultName;
+        counter = new AtomicLong();
+        this.jerseyClient = jerseyClient;
+        this.blobStore = blobStore;
+    }
+
+    public Client getJerseyClient() {
+        return jerseyClient;
+    }
+
+    @GET
+    @Timed
+    @Consumes(MediaType.APPLICATION_JSON)
+    public ClientResponse getMessageFromServer(@QueryParam("name") String name) {
+
+        final String clientName = name != null ? name : defaultName;
+        final String message = String.format(template, clientName);
+        final ClientRequest clientRequest = new ClientRequest(clientName, message);
+
+        BlobContext blobContext = null;
+        if (blobStore != null) {
+            blobContext = new SimpleBlobContext("ServerService", "getMessageFromServer");
+
+            Blobs requestBlob = createBlob(createBlobFactory(), blobContext);
```

Comment:
BlobFactory should be a singleton. Should be created only once when the app starts and passed. Each request should execute only ```blobsFactory.create(blobContext);```

Reply:
Made the changes. Please view the latest commit.